src/slash/database.ts

```diff
@@ -90,25 +86,88 @@ export async function authenticatePlayer(
+		new ButtonBuilder()
+			.setCustomId(encodeCustomId("accept", tournament.tournamentId))
+			.setLabel("Accept")
+			.setStyle(ButtonStyle.Success),
+		new ButtonBuilder()
+			.setCustomId(encodeCustomId("reject", tournament.tournamentId))
+			.setLabel("Reject")
+			.setStyle(ButtonStyle.Danger)
 	);
 	return row;
 }
 
-async function awaitModalInteraction(
-	modalInteraction: ModalSubmitInteraction<CacheType>,
-	deck: ManualDeckSubmission,
-	tournament: ManualTournament,
-	response: Message
-): Promise<void> {
-	const player = await response.client.users.fetch(deck.discordId);
-	if (modalInteraction.customId === "acceptModal") {
+export class AcceptButtonHandler implements ButtonClickHandler {
+	readonly buttonIds = ["accept"];
+
+	async click(interaction: ButtonInteraction, ...args: string[]): Promise<void> {
+		const tournamentIdString = args[0];
+		const modal = new ModalBuilder()
+			.setCustomId(encodeCustomId("acceptModal", tournamentIdString))
+			.setTitle("Accept Deck");
+		const deckLabelInput = new TextInputBuilder()
+			.setCustomId("acceptDeckLabel")
+			.setLabel("What is the deck's theme?")
+			.setStyle(TextInputStyle.Short)
+			.setRequired(false);
+		const actionRow = new ActionRowBuilder<TextInputBuilder>().addComponents(deckLabelInput);
+		modal.addComponents(actionRow);
+		await interaction.showModal(modal);
+	}
+}
+
+export class RejectButtonHandler implements ButtonClickHandler {
+	readonly buttonIds = ["reject"];
+
+	async click(interaction: ButtonInteraction, ...args: string[]): Promise<void> {
+		const tournamentIdString = args[0];
+		const modal = new ModalBuilder()
+			.setCustomId(encodeCustomId("rejectModal", tournamentIdString))
+			.setTitle("Reject Deck");
+		const rejectReasonInput = new TextInputBuilder()
+			.setCustomId("rejectReason")
+			.setLabel("Why is the deck illegal?")
+			.setStyle(TextInputStyle.Paragraph)
+			.setRequired(false);
+		const actionRow = new ActionRowBuilder<TextInputBuilder>().addComponents(rejectReasonInput);
+		modal.addComponents(actionRow);
+		await interaction.showModal(modal);
+	}
+}
+
+export class AcceptLabelModal implements MessageModalSubmitHandler {
+	readonly modalIds = ["acceptModal"];
+
+	async submit(interaction: ModalMessageModalSubmitInteraction, ...args: string[]): Promise<void> {
+		const tournamentIdString = args[0];
+		const deck = await ManualDeckSubmission.findOneOrFail({
+			where: {
+				discordId: interaction.user.id,
+				tournamentId: parseInt(tournamentIdString, 10)
+			},
+			relations: ["tournament"]
+		});
+		const tournament = deck.tournament;
+		const player = await interaction.client.users.fetch(deck.discordId);
 		// mark deck as approved
 		deck.approved = true;
-		const label = modalInteraction.fields.getTextInputValue("acceptDeckLabel");
+		const label = interaction.fields.getTextInputValue("acceptDeckLabel");
 		if (label.length > 0) {
 			deck.label = label;
 		}
```

Comment:
So the line that fails is the one below this one

Reply:
Right, because that's the one that actually commits the changes to the database. But we're editing an already-existing entry, why does the tournamentId suddenly become null?